### ISSUE TITLE
Integer overflow on track length

### ISSUE THREAD

**Issue by RollingStar** [OpenRCT2/OpenRCT2#1013]
http://www.reddit.com/r/rct/comments/32ev9w/2_i_decided_to_make_a_long_ride_but_i_made_it_so/

**Comment by kevinburke** [comment]
Gonna be tricky to fix so long as anything in the game reads from the old TD6 format :( The game reads from specific byte offsets in that format and upgrading the length would prolly break it.

**Comment by RomanovX** [comment]
Agree, the best course of action IMHO is to possibly reevaluate TD6 once the decompiling is done. But until then, keep those limitations.

**Comment by Gymnasiast** [comment]
TD6 is used to save track designs separately. Its structure is not the same as that of the rides in a SV6 or SC6. 
Also, it doesn't say whether the same happens in vanilla RCT2.

**Comment by janisozaur** [comment]
There were some changes to handling TD6 recently, this should be checked now.

**Comment by rwjuk** [comment]
This still occurs as far as I can tell, an SV6 demonstrating this is attached.

[10x10 max length experiment.sv6.zip](https://github.com/OpenRCT2/OpenRCT2/files/1030269/10x10.max.length.experiment.sv6.zip)

**Comment by Gymnasiast** [comment]
Oh, this is interesting. It only displays negative numbers if your game is set to Olde English measurements (Imperial).

**Comment by Gymnasiast** [comment]
Reintroduced by #13646. Needs changing length variables to 32 bits.

**Comment by spacek531** [comment]
Is this bug still occurring? Using the sv6 above the ride tells me it has a length of 54,061 ft/16,476 m. Are these the expected results?

**Comment by 733737** [comment]
If it isn't negative it is probably fixed

**Comment by tupaschoal** [comment]
@Gymnasiast you were able to reproduce before, can you please check?

**Comment by Gymnasiast** [comment]
I can no longer reproduce this, assuming it has been fixed.
